### PR TITLE
Unit Testing for quicker testing

### LINKED ISSUE #24: Offline Mock Unit Testing
While we can't really use them to test with real datastores, pushing any logic updates we can easily test using tests that use every method.

### PR DESCRIPTION
**Pull request by noahrepublic**
Began creating the unit testing workflow for quickly checking if any updates lose functionality.

### CODE REVIEW

**Comment by noahrepublic** [comment]
once its done we will be clearing the testing/server directory

**Comment by noahrepublic** [comment]
Need to write tests for locking, clearing, listening for globals, etc.

**Comment by noahrepublic** [comment]
Seems mostly done. However, want to wait on: https://github.com/noahrepublic/DataKeep/issues/27

File: src/MockStore/init.lua [hunk 1]
```diff
@@ -84,6 +87,10 @@ function MockStore:UpdateAsync(key: string, callback: (any) -> any)
 	local value = self._data[key]
 	local yielded, newValue = didyield(callback, value)
 
+	-- print("Update async called")
```
Comment (FireBird702) [inline src/MockStore/init.lua]: Maybe delete these comments

Comment (noahrepublic) [reply src/MockStore/init.lua]: My bad, I have a habit of leaving debug prints in. Will do

File: test/wally.toml
Comment (FireBird702) [inline test/wally.toml:1]: Why not add [dev-dependencies] to main wally.toml which will install all dependencies during testing?

File: wally.lock
Comment (FireBird702) [inline wally.lock:1]: The .lock file can be removed from repo. I see that is should be ignored based on what .gitignore says but somehow its still there

File: src/init.spec.lua [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+		-- it("should not let viewkeeps change data", function()
+		-- 	local viewKeep = testStore:ViewKeep("Data"):expect()
+
+		-- 	viewKeep.Data.Coins = 100
+		-- 	testStore.IssueSignal:Connect(function(err)
+		-- 		expect(err).to.be.ok()
+		-- 	end)
+
+		-- 	viewKeep:Save() -- not sure how to catch this. this doesn't error but processError does.
```
Comment (FireBird702) [inline src/init.spec.lua:156]: I was thinking about `:Save()`, `:Overwrite()` and `:Release()` rejecting. Maybe next goal after this pr gets merged?

**Review by FireBird702 — CHANGES_REQUESTED** [review]

File: wally.lock
Comment (noahrepublic) [reply wally.lock:1]: Yeah unsure how to resolve that.

Comment (FireBird702) [reply wally.lock:1]: Delete it and commit the deletion. Maybe it will disappear

Comment (noahrepublic) [reply wally.lock:1]: Tried, seems has more steps to it we also have a .lock in the main wally

File: src/init.spec.lua (hunk 2 above)
Comment (noahrepublic) [reply src/init.spec.lua:156]: Good idea

File: .gitignore [hunk 3]
```diff
@@ -10,3 +10,10 @@
 /sourcemap.json
 
 /Packages
+/DevPackages
+
+
+/test/Packages
```
Comment (FireBird702) [inline .gitignore]: This can be deleted because there is no /test folder after changing dev-dependencies
Besides that, it looks good for me

**Review by FireBird702 — APPROVED** [review]

File: .gitignore (hunk 3 above)
Comment (noahrepublic) [reply .gitignore]: /test/Packages yes, but DevPackages we need, since dev-dependencies goes into devpackages

Comment (FireBird702) [reply .gitignore]: DevPackages are stored in main directory

**Comment by noahrepublic** [comment]
Good to merge?

**Comment by FireBird702** [comment]
Yes

**Pull request merged**